src/data/nat/periodic.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2021 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+import algebra.periodic
+import data.nat.interval
+import data.nat.count
+
+/-!
+# Periodic Functions on ℕ
+
+This file identifies a few functions on `ℕ` which are periodic, and also proves a lemma about
+periodic predicates which helps determine their cardinality when filtering intervals over them.
+-/
+
+open finset nat function
+
+lemma periodic_gcd (a : ℕ) : periodic (gcd a) a := by simp
```

Comment:
I think that in the near future, `periodic` should be a def that is not tagged with `@[simp]`.

Reply:
Let me change these to `simp only` then.